Source/TextKit/ASTextKitCoreTextAdditions.m

```diff
@@ -227,15 +233,33 @@ + (instancetype)paragraphStyleWithCTParagraphStyle:(CTParagraphStyleRef)coreText
   CGFloat minimumLineHeight;
   if (CTParagraphStyleGetValueForSpecifier(coreTextParagraphStyle, kCTParagraphStyleSpecifierMinimumLineHeight, sizeof(minimumLineHeight), &minimumLineHeight))
     newParagraphStyle.minimumLineHeight = minimumLineHeight;
-
-  // kCTParagraphStyleSpecifierLineSpacing -> lineSpacing
-  // Note that kCTParagraphStyleSpecifierLineSpacing is deprecated and will die soon. We should not be using it.
+  
+  CGFloat lineSpacing = 0;
+#if TARGET_OS_IOS
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-    CGFloat lineSpacing;
-    if (CTParagraphStyleGetValueForSpecifier(coreTextParagraphStyle, kCTParagraphStyleSpecifierLineSpacing, sizeof(lineSpacing), &lineSpacing))
-        newParagraphStyle.lineSpacing = lineSpacing;
+  // kCTParagraphStyleSpecifierLineSpacing -> lineSpacing
+  // Note that kCTParagraphStyleSpecifierLineSpacing is deprecated and will die soon. We should not be using it.
+  if (CTParagraphStyleGetValueForSpecifier(coreTextParagraphStyle, kCTParagraphStyleSpecifierLineSpacing, sizeof(lineSpacing), &lineSpacing))
+    newParagraphStyle.lineSpacing = lineSpacing;
 #pragma clang diagnostic pop
+#endif
+  
+  // Attempt to weakly map the following onto -[NSParagraphStyle lineSpacing]:
+  //   - kCTParagraphStyleSpecifierMinimumLineSpacing
+  //   - kCTParagraphStyleSpecifierMaximumLineSpacing
+  //   - kCTParagraphStyleSpecifierLineSpacingAdjustment
+  if (fabs(lineSpacing) <= FLT_EPSILON &&
+      CTParagraphStyleGetValueForSpecifier(coreTextParagraphStyle, kCTParagraphStyleSpecifierMinimumLineSpacing, sizeof(lineSpacing), &lineSpacing))
+    newParagraphStyle.lineSpacing = lineSpacing;
```

Comment:
All if statements should use braces, even for one line. I'd suggest doing that for these statements even if the rest of the file doesn't / did not previously, since the conditions are so long.

Reply:
Maybe worth making a macro for these checks, or at least a local variable for the FLT_EPSILON check.